src/world/Manager/MapMgr.cpp

```diff
@@ -0,0 +1,889 @@
+    {
+      if( count++ % 100 == 0 )
+        std::cout << ".";
+
+      auto level = exdData.get< Data::Level >( issuerLevelId );
+      auto territory = level->territory;
+
+      if( territory == 0 )
+        territory = exdData.get< Data::Map >( level->map )->territoryType;
+
+      EventData eventData;
+      eventData.iconId = questData.iconValid;
+      eventData.levelId = issuerLevelId;
+      eventData.actorId = questData.issuer;
+
+      m_mapData[ territory ].emplace( quest.first, eventData );
+    }
+  }
+  
+  auto& serverMgr = Common::Service< World::ServerMgr >::ref();
+  auto m_gameData = new xiv::dat::GameData( serverMgr.getConfig().global.general.dataPath );
+
+  std::set< std::string > bgSet;
+
+  for( auto territoryTypeId : exdData.getTerritoryTypeIdList() )
+  {
+    auto territoryType = exdData.get< Data::TerritoryType >( territoryTypeId );
+    if( territoryType->bg.empty() )
+      continue;
+
+    if( bgSet.find( territoryType->bg ) != bgSet.end() )
+      continue;
+    else
+      bgSet.insert( territoryType->bg );
+
+    std::string planeventLgbPath( "bg/" + territoryType->bg.substr( 0, territoryType->bg.rfind( '/' ) ) + "/planevent.lgb" );
+    auto planeventFile = m_gameData->getFile( planeventLgbPath );
+    auto planeventData = planeventFile->access_data_sections().at( 0 );
+    LGB_FILE planeventLgb( &planeventData[ 0 ], "planevent" );
+
+    for( const auto& group : planeventLgb.groups )
+    {
+      for( const auto& pEntry : group.entries )
+      {
+        if( pEntry->getType() == LgbEntryType::EventNpc )
+        {
+          auto pNpc = reinterpret_cast< LGB_ENPC_ENTRY* >( pEntry.get() );
+
+          auto eNpcData = exdData.get< Data::ENpcBase >( pNpc->data.enpcId )->eNpcData;
+
+          for( auto npcData : eNpcData )
+          {
+            if( npcData == 0 )
+              continue; // Some npcs have data gaps, so we have to iterate through the entire array
+
+            if( count++ % 1000 == 0 )
+              std::cout << ".";
+
+            EventData eventData;
+            eventData.levelId = pNpc->data.instanceId;
+            eventData.actorId = pNpc->data.enpcId;
+
+            auto eventHandlerType = static_cast< EventHandler::EventHandlerType >( npcData >> 16 );
+
+            switch( eventHandlerType )
+            {
+              case EventHandler::EventHandlerType::GuildLeveAssignment:
+              {
+                eventData.iconId = exdData.get< Data::EventIconType >( 5 )->mapIconAvailable + 1;
+
+                m_mapData[ territoryTypeId ].insert( std::make_pair( npcData, eventData ) );
+                break;
+              }
+
+              case EventHandler::EventHandlerType::CustomTalk:
+              {
+                // Include only the beginner arena icon yet. There a few other ones, that aren't referenced in the game files (Some examples are: The Triple Triad Tournament npc which has multiple icons and the ocean fishing icon)
+                if( npcData == 721223 )
```

Comment:
Sorry, but I do not understand where the problem here is.
The custom talk icons have a very special handling from the server and not all of them are referenced in the exd data (The server decides which icon is displayed on the map. This applies to all supported events). So I have included this icon for now only.

Reply:
i think i started writing something a few days ago and then didn't finish, i think what i intended to write here is that this shouldn't be hardcoded in some random file but i can't be sure lol

Reply:
Aah, you're right, that this should'nt be hardcoded, but I think this will do it for now, until it is figured out which custom talk icons are shown by retail. In addition it is unclear for me now which icons are displayed for custom talk npcs that aren't referenced in the exd data.